jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/LoggingMXBeanImpl.java

```diff
@@ -151,20 +230,61 @@ public String getParentLoggerName(String loggerName) {
 	 */
 	@Override
 	public void setLoggerLevel(String loggerName, String levelName) {
-		final Logger logger = LogManager.getLogManager().getLogger(loggerName);
-
+		final Object logger;
+		try {
+			logger = getLoggerFromName(loggerName);
+		} catch (Exception e) {
+			throw newInternalInvokeError(e);
+		}
+		
 		if (logger != null) {
 			// The named Logger exists. Now attempt to set its level. The
 			// below attempt to parse a Level from the supplied levelName
 			// will throw an IllegalArgumentException if levelName is not
 			// a valid level name.
-			Level newLevel = Level.parse(levelName);
-			logger.setLevel(newLevel);
+			try {
+				Object newLevel = level_parse.invoke(null, levelName);
+				logger_setLevel.invoke(logger, newLevel);
+			} catch (Exception e) {
+				throw newInternalInvokeError(e);
+			}
+			
 		} else {
 			// Named Logger does not exist.
 			/*[MSG "K05E7", "Unable to find Logger with name {0}."]*/
 			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K05E7", loggerName)); //$NON-NLS-1$
 		}
 	}
-
+	
+	private static Object getLoggerFromName(String loggerName) throws Exception {
+		Object logManagerInstance = logManager_getLogManager.invoke(null);
+		return logManager_getLogger.invoke(logManagerInstance, loggerName);
+	}
+	
+	/* Handle error thrown by method invoke as internal error */
+	private static InternalError newInternalInvokeError(Exception cause) {
+		handleInvocationTargetException(cause);
+		
+		InternalError err = new InternalError(cause.toString());
+		err.initCause(cause);
+		throw err;
+	}
+	
+	/* invoke throws InvocationTargetException if the method it is invoking
+	 * throws an error. This method unwraps that error for this class to maintain
+	 * its specification.
+	 */
+	private static void handleInvocationTargetException(Exception cause) {
+		if (cause instanceof InvocationTargetException) {
+			Exception wrappedCause = (Exception)cause.getCause();
+			
+			if (wrappedCause instanceof SecurityException) {
```

Comment:
This could be more generic and check for RuntimeException and Error, as any subclasses of these can be re-thrown. Might avoid a problem in the future if the specification changes to add another RuntimeException.